File: Backend/models/contract.py
from enum import Enum
from typing import Optional
from pydantic import BaseModel, EmailStr

class ContractStatus(str, Enum):
    REQUESTED = 'solicitado'
    REJECTED = 'rechazado'
    IN_PROGRESS = 'en_progreso'
    COMPLETED = 'completado'
    
class Contract(BaseModel):
    _id: Optional[str] = None
    client_id: str
    service_id: str
    description: str
    status: ContractStatus = ContractStatus.REQUESTED
    
    

File: web/src/app/models/contract.ts
export enum ContractStatus {
    REQUESTED = 'solicitado',
    REJECTED = 'rechazado',
    IN_PROGRESS = 'en_progreso',
    COMPLETED = 'completado',
    CANCELED = 'cancelado'
}
export interface ContractModel {
    _id?: string
    client_id?: string
    service_id?: string
    provider_id?: string 
    description: string
    status: ContractStatus | string
    provider_name?: string
    service_name?: string
    client_name?: string
    request_date?: string
}